File: clang/test/Modules/linkage-merge.m
// RUN: rm -rf %t
// RUN: %clang_cc1 -fmodules -x objective-c -fmodules-cache-path=%t -emit-module -fmodule-name=linkage_merge_left %S/Inputs/module.map
// RUN: %clang_cc1 -fmodules -fmodules-cache-path=%t -w %s -verify

// Test redeclarations of functions where the original declaration is
// still hidden.

@import linkage_merge_left; // excludes "sub"

extern int f0(float);
// expected-error@-1{{conflicting types for 'f0'}}
// expected-note@Inputs/linkage-merge-sub.h:1{{previous declaration}}

static int f1(float); // okay: considered distinct
static int f2(float); // okay: considered distinct
extern int f3(float); // okay: considered distinct

extern float v0;
// expected-error@-1{{redefinition of 'v0' with a different type: 'float' vs 'int'}}
// expected-note@Inputs/linkage-merge-sub.h:6{{previous definition is here}}

static float v1;
static float v2;
extern float v3;

typedef float T0;

File: clang/test/Modules/linkage-merge.cpp
// RUN: rm -rf %t
// RUN: %clang_cc1 -verify -fmodules -fimplicit-module-maps -fmodules-cache-path=%t -I %S/Inputs %s

#include "linkage-merge-bar.h"

static int f(int);
int f(int);

static void g(int);
// FIXME: Whether we notice the problem here depends on the order in which we
// happen to find lookup results for 'g'; LookupResult::resolveKind needs to
// be taught to prefer a visible result over a non-visible one.
//
// expected-error@9 {{functions that differ only in their return type cannot be overloaded}}
// expected-note@Inputs/linkage-merge-foo.h:2 {{previous declaration is here}}
